force-app/main/default/classes/DMLOperations.cls

```diff
@@ -148,6 +166,35 @@ public with sharing class DMLOperations {
         // Loop through each Opportunity name
             // Create an Opportunity with the given name
             // Add the Opportunity to the list
+        List<Account> accounts = [SELECT Id FROM Account WHERE Name = :accountName];
+        if(accounts.isEmpty()){
+            Account newAcc = new Account(Name = accountName);
+            Database.insert(newAcc, AccessLevel.SYSTEM_MODE);
+            accounts.add(newAcc);
+        } 
+        List<Opportunity> opportunitiesThatExist = [SELECT Name, Id FROM Opportunity WHERE Name IN :oppNames AND AccountId = :accounts[0].Id];
+
+        for(String oppName: oppNames){
+            Boolean found = false;
+                for(Opportunity opp: opportunitiesThatExist){
+                    if(opp.Name.compareTo(oppName) == 0){
+                        found = true;
+                        opp.AccountId = accounts[0].Id;
+                        break;
+                    }
+                }
+            if(found == false){
```

Comment:
Instead of using a found variable check use the opportunity object. Check if it's null or not.